enhancements/console/expose-network-features.md

```diff
@@ -0,0 +1,168 @@
+---
+title: console-expose-network-features
+authors:
+- "@mariomac"
+- "@jotak"
+reviewers:
+- "@abhat"
+- "@jotak"
+approvers:
+- "@abhat"
+creation-date: 2021-07-07
+last-updated: 2021-07-07
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Console: expose network features
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Console may need to show some fields that are exclusive to a given CNI type, and hide the fields that are not
+provided by it. Then the console needs visibility about which CNI (`OpenShiftSDN` or `OVNKubernetes`) features
+are available, depending on each customer's configuration.
+
+This information should be available to all users (or, at this moment, to any user able to create
+a Network Policy).
+
+## Motivation
+
+[In the new network policy creation forms, some fields may depend on the type
+of cluster network type](https://issues.redhat.com/browse/NETOBSERV-16).
+For example, `OpenShiftSDN` neither supports egress
+network policies nor ingress exceptions. The related fields should be only visible
+when the cluster network is `OVNKubernetes`.
+
+### Goals
+
+* The console can fetch a set of CNI features, which would depend on the CNI type, whichever user
+  is logged in.
+
+### Non-Goals
+
+N/A.
+
+## Proposal
+
+The Console UI should be able to read the set of features that the SDN component supports. These
+features are accessed via a ConfigMap (read-only permissions) that is written by the Cloud Network
+Operator (CNO). For example, the CNO would write a Config Map with a list of features, e.g.:
+
+```properties
+network_policy_egress=true
```

Comment:
Is this the full list of features you want to be exposed per CNI currently?

Reply:
yes, at the moment.